src/Servant/Elm/Internal/Generate.hs

```diff
@@ -450,15 +465,63 @@ isEmptyType :: ElmOptions -> ElmDatatype -> Bool
 isEmptyType opts elmTypeExpr =
   elmTypeExpr `elem` emptyResponseElmTypes opts
 
+toStringSrc :: T.Text -> ElmOptions -> ElmDatatype -> Doc
+toStringSrc operator opts argType
+  -- Don't use "toString" on Elm Strings, otherwise we get extraneous quotes.
+  | isElmStringType opts argType = stext ""
+  | otherwise                    = stext $ toStringSrcTypes operator opts argType <> " " <> operator
+
 
 {- | Determines whether we call `toString` on URL captures and query params of
 this type in Elm.
 -}
 isElmStringType :: ElmOptions -> ElmDatatype -> Bool
+isElmStringType _ (ElmPrimitive (EList (ElmPrimitive EChar))) = True
 isElmStringType opts elmTypeExpr =
   elmTypeExpr `elem` stringElmTypes opts
 
 
+toStringSrcTypes :: T.Text -> ElmOptions -> ElmDatatype -> T.Text
+toStringSrcTypes operator opts (ElmPrimitive (EMaybe argType)) = toStringSrcTypes operator opts argType
+toStringSrcTypes _ _ (ElmPrimitive (EList (ElmPrimitive EChar))) = "identity"
```

Comment:
It seems like this case would give a different type from the other cases, but I don't really understand how this function is supposed to work 😅. What does it do?

Reply:
well document.

Reply:
*will